admin/controllers/pupil.js

```diff
@@ -95,27 +82,142 @@ const postAddPupil = async (req, res, next) => {
   } catch (error) {
     next(error)
   }
-  res.redirect(`/school/pupil-register/lastName/true?hl=${pupil._id}`)
+  const pupilId = JSON.stringify([pupil._id])
+  res.redirect(`/school/pupil-register/lastName/true?hl=${pupilId}`)
 }
 
-const getAddMultiplePupils = async (req, res, next) => {
+const getAddMultiplePupils = (req, res, next) => {
   res.locals.pageTitle = 'Add multiple pupils'
+  const { hasError } = res
   try {
     req.breadcrumbs('Pupil Register', '/school/pupil-register/lastName/true')
     req.breadcrumbs(res.locals.pageTitle)
     res.render('school/add-multiple-pupils', {
-      breadcrumbs: req.breadcrumbs()
+      breadcrumbs: req.breadcrumbs(),
+      hasError
     })
   } catch (error) {
     next(error)
   }
 }
 
+const postAddMultiplePupils = async (req, res, next) => {
+  let school
+  try {
+    school = await School.findOne({_id: req.user.School}).exec()
+    if (!school) {
+      throw new Error(`School [${req.body.school}] not found`)
+    }
+  } catch (error) {
+    return next(error)
+  }
+  const uploadFile = req.files.csvTemplateFile
+  let csvData = []
+  const stream = fs.createReadStream(uploadFile.file)
+  const csvStream = csv()
+    .on('data', (data) => { csvData.push(data) })
+    .on('end', async() => {
+      // Remove error column and headers from data
+      if (csvData.some(p => p[6])) csvData.map((r) => r.splice(6, 1))
+      let headers = csvData.shift(0)
+      // validate each pupil
+      const pupils = []
+      csvData = await Promise.all(csvData.map(async(p) => {
+        const pupil = new Pupil({
+          school: school._id,
+          upn: p[3],
+          foreName: p[0],
+          lastName: p[2],
+          middleNames: p[1],
+          gender: p[5],
+          dob: p[4],
+          pin: null,
+          pinExpired: false
+        })
+        try {
+          const dob = p[4].split('/')
+          const pupilData = Object.assign({
+            'dob-day': dob[1],
+            'dob-month': dob[0],
+            'dob-year': dob[2]
+          }, pupil._doc)
+          await validatePupil(pupil, pupilData)
+        } catch (err) {
+          p[6] = []
+          Object.keys(err.errors).forEach((e) => p[6].push(err.errors[e].replace(/ \\"/g, '')))
```

Comment:
Why even have the smart quotes in the error message if you remove them?

Reply:
The double quotes within single quotes causes an issue on Excel. These content within the double quotes will appear as an ascii representation which is obviously unreadable.

Reply:
Can you provide an example of the output that is causing an issue?  Is it only Excel that has the issue?

Reply:
So far I only have noticed it on Excel.
'Please check “Year”' will appear as Please check â€œYearâ€

Reply:
It should be fine in UTF-8.... however it may be more accessible if you use only ASCII characters.  Does it work if you use normal straight double-quotes rather than curlies?